### PR TITLE
Update h3

### PR DESCRIPTION
**Pull request by qoega**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

...

Detailed description / Documentation draft:

...

By adding documentation, you'll allow users to try your new feature immediately, not when someone else will have time to document it later. Documentation is necessary for all features that affect user experience in any way. You can add brief documentation draft above, or add documentation right into your patch as Markdown files in [docs](https://github.com/ClickHouse/ClickHouse/tree/master/docs) folder.

If you are doing this for the first time, it's recommended to read the lightweight [Contributing to ClickHouse Documentation](https://github.com/ClickHouse/ClickHouse/tree/master/docs/README.md) guide first.

Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by qoega** [comment]
FYI @bharatnc

File: src/Functions/geoToH3.cpp [hunk 1]
```diff
@@ -79,11 +80,17 @@ class FunctionGeoToH3 : public IFunction
...
-            coord.lon = degsToRads(lon);
+            LatLng coord;
+            coord.lng = degsToRads(lon);
             coord.lat = degsToRads(lat);
-
-            H3Index hindex = geoToH3(&coord, res);
+ 
+	    H3Index hindex;
+	    H3Error err = latLngToCell(&coord, res, &hindex);
+	    if (err) {
```
Comment (bharatnc) [inline src/Functions/geoToH3.cpp:89]:
```suggestion
	    if (err) 
	    {
```

**Comment by alexey-milovidov** [comment]
@mergifyio update

**Review by alexey-milovidov — APPROVED** [review]

**Comment by bharatnc** [comment]
@alexey-milovidov @qoega 

- It looks like after these latest changes, the submodule for h3 is no longer in sync with what we have in ClickHouseExtras/h3 https://github.com/ClickHouse-Extras/h3. Should ClickHouse-Extras/h3. also be updated ?

- It also looks like after these changes the h3 apis available to us conform to only [4.x(alpha)](https://h3geo.org/docs/next/api/indexing) and not [3.x(stable)]((https://h3geo.org/docs/api/indexing). Example compare traversal funcs available b/w the two versions, the list is very different (it looks to me like there have been a lot of changes to function names and apis exposed).

- Does this mean that we should only be following the api reference for 4.x (alpha) ? Related to #/17708 where I'm trying to gather a list of functions yet to be supported in ClickHouse and I'm planning to start adding them.

**Pull request merged**